### DISCUSSION TITLE
[NOT AN ISSUE] Update checker

### DISCUSSION THREAD

**Discussion by fdicarlo** [maester365/maester#330] [General]
Thinking about the logic of Maester and/or the Slack bot - would it make sense to have an update checker to see when a new Maester version is released? The team using Maester may not always checking GitHub.

I tried to script something but I am not sure about it:

```
    - name: Checkout repository
      uses: actions/checkout@v2

    - name: Get latest release
      id: get_release
      run: |
        latest_release=$(curl -s https://api.github.com/repos/maester365/maester/releases/latest | jq -r .tag_name)
        echo "latest_release=${latest_release}" >> $GITHUB_ENV

    - name: Compare with last checked release
      id: compare_release
      run: |
        last_release=$(cat .last_release 2>/dev/null || echo "none")
        echo "Last checked release: ${last_release}"
        echo "Latest release: ${latest_release}"
        if [ "${latest_release}" != "${last_release}" ]; then
          echo "New release found: ${latest_release}"
          echo "new_release=true" >> $GITHUB_ENV
          echo "${latest_release}" > .last_release
        else
          echo "No new release found"
          echo "new_release=false" >> $GITHUB_ENV

    - name: Notify Slack about new release
      if: env.new_release == 'true'
      run: |
        slack_message="New release detected: ${latest_release}. See details: https://github.com/maester365/maester/releases/tag/${latest_release}"
        curl -X POST -H 'Content-type: application/json' --data "{\"text\":\"${slack_message}\"}" "${{ secrets.SLACK_WEBHOOK_URL }}"
```

**Comment by SamErde** [comment]
The `Invoke-Maester`, `Install-MaesterTests`, and `Update-MaesterTests` functions all call the internal check `Get-IsNewMaesterVersionAvailable | Out-Null`. I wonder if there was a plan to return to this and prompt the user to update if a new version is found. @merill or @f-bader may know?

**Comment by merill** [comment]
@SamErde  I personally am not a fan when apps force me to update when I'm trying to do something. So the warning message was a way to let the user be in control and they can run the update if they wanted to, without affecting their flow.

This is usually the behavior in other CLI tools like homebrew and even PowerShell itself.

This is not really a big issue for folks that set up automation, since each run the module gets installed so the latest version from the gallery is used. 

@fdicarlo I do love that you've already written the code for Slack, we can always add it to the docs for those who wish to set it up. Do you want to create a PR for the docs with this?

**Comment by fdicarlo** [comment]
> @SamErde I personally am not a fan when apps force me to update when I'm trying to do something. So the warning message was a way to let the user be in control and they can run the update if they wanted to, without affecting their flow.
> 
> This is usually the behavior in other CLI tools like homebrew and even PowerShell itself.
> 
> This is not really a big issue for folks that set up automation, since each run the module gets installed so the latest version from the gallery is used.
> 
> @fdicarlo I do love that you've already written the code for Slack, we can always add it to the docs for those who wish to set it up. Do you want to create a PR for the docs with this?

Hi @merill I agree, I don't like forced updates either hence why it was just an update checker, not an enforcer :)
About the code, I wrote it but not tested enough, let me do few other tests and come back on it

**Comment by merill** [comment]
Cheers @fdicarlo 

I'll move this to a discussion.

**Comment by SamErde** [comment]
@fdicarlo and @merill, I agree completely with your preference to not force the user to update their app. That was not the intent of my comments, just to be clear. 😄

My question was really regarding the line of code used in several places that executes `Get-IsNewMaesterVersionAvailable | Out-Null`. I had assumed `Out-Null` would suppress the output of `Get-IsNewMaesterVersionAvailable`. However, after looking more closely at that function, I realized that it just suppresses the Boolean that gets returned, and not the messages written to the host during the execution of `Get-IsNewMaesterVersionAvailable`. 👍
